lib/Assertion/ExpressionEvaluatorFactory.php

```diff
@@ -2,15 +2,23 @@
 
 namespace PhpBench\Assertion;
 
-use PhpBench\Assertion\MessageFormatter\NodeMessageFormatter;
-
 final class ExpressionEvaluatorFactory
 {
+    /**
+     * @var ExpressionFunctions
+     */
+    private $functions;
+
+    public function __construct(ExpressionFunctions $functions)
+    {
+        $this->functions = $functions;
+    }
+
     /**
      * @param array<string,mixed> $args
      */
     public function createWithArgs(array $args): ExpressionEvaluator
     {
-        return new ExpressionEvaluator(new NodeMessageFormatter($args), $args);
+        return new ExpressionEvaluator($args, $this->functions);
     }
```

Comment:
Can refactor to pass `Node $node, array $args` rather than injet the args in the ctor and use a factory.